global.php

```diff
@@ -111,21 +111,36 @@
 // The user used our new quick theme changer
 if(isset($mybb->input['theme']) && verify_post_check($mybb->get_input('my_post_key'), true))
 {
-	$mybb->user['style'] = $mybb->get_input('theme');
-	// If user is logged in, update their theme selection with the new one
-	if($mybb->user['uid'])
+	// Set up user handler.
+	require_once MYBB_ROOT.'inc/datahandlers/user.php';
+	$userhandler = new UserDataHandler('update');
+
+	$user = array(
+		'uid'	=> $mybb->user['uid'],
+		'style'	=> $mybb->get_input('theme', MyBB::INPUT_INT)
+	);
+
+	$userhandler->set_data($user);
+
+	if($userhandler->verify_style())
 	{
-		if(isset($mybb->cookies['mybbtheme']))
+		$mybb->user['style'] = $user['style'];
+
+		// If user is logged in, update their theme selection with the new one
+		if($mybb->user['uid'])
 		{
-			my_unsetcookie('mybbtheme');
-		}
+			if(isset($mybb->cookies['mybbtheme']))
+			{
+				my_unsetcookie('mybbtheme');
+			}
 
-		$db->update_query('users', array('style' => (int)$mybb->user['style']), "uid = '{$mybb->user['uid']}'");
-	}
-	// Guest = cookie
-	else
-	{
-		my_setcookie('mybbtheme', $mybb->get_input('theme'));
+			$userhandler->update_user();
+		}
+		// Guest = cookie
+		else
+		{
+			my_setcookie('mybbtheme', $mybb->get_input('theme'));
+		}
 	}
 }
 // Cookied theme!
```

Comment:
Exactly how do you want to verify the cookie? On every page load seems the correct but not best approach.